app/models/volunteer_op.rb

```diff
@@ -34,4 +42,16 @@ def self.contains_description?(text)
   def self.contains_title?(text)
     arel_table[:title].matches(text)
   end
+  
+  def address_complete?
+    address.present? && postcode.present?
+  end 
+  
+  private
+  
+  def clear_lat_lng
+    return if address_complete?
+    self.longitude = nil
+    self.latitude = nil
+  end
```

Comment:
We've still got a set of failing tests in CI, specifically:

```
undefined method `latitude' for nil:NilClass (NoMethodError)
./app/services/build_markers_with_info_window.rb:24:in `build_single_marker'
./app/services/build_markers_with_info_window.rb:20:in `run'
./app/services/build_markers_with_info_window.rb:4:in `with'
./app/controllers/volunteer_ops_controller.rb:18:in `index'
features/volunteer_opportunities/doit_volunteer_opportunities.feature:45:in `And the map should show the do-it opportunity titled Cub Scout Leader (volunteering with 8-10 year olds) 27th Harrow'

No route matches {:action=>"show", :controller=>"volunteer_ops", :id=>nil} (ActionController::UrlGenerationError)
./features/step_definitions/navigation_steps.rb:91:in `/^I (visit|should be on) the (edit|show) page for the (.*?) (named|titled) "(.*?)"$/'
features/volunteer_opportunities/show_volunteer_opportunity.feature:33:in `Given I visit the show page for the volunteer_op titled "Mini Bus Driver"'

Failing Scenarios:
cucumber features/volunteer_opportunities/doit_volunteer_opportunities.feature:12 # Scenario: See a map of current doit volunteer opportunities
cucumber features/volunteer_opportunities/doit_volunteer_opportunities.feature:16 # Scenario: See a map of current doit and harrow volunteer opportunities
cucumber features/volunteer_opportunities/doit_volunteer_opportunities.feature:31 # Scenario: See a list of current doit volunteer opportunities with a link to  organisation page
cucumber features/volunteer_opportunities/doit_volunteer_opportunities.feature:37 # Scenario: See a list of current doit volunteer opportunities with a link to opportunity page
cucumber features/volunteer_opportunities/doit_volunteer_opportunities.feature:43 # Scenario: Doit volunteer opportunites are listed in map popups
cucumber features/volunteer_opportunities/show_volunteer_opportunity.feature:32 # Scenario: See volunteer opportunity location with different address
```

I am also getting them locally - is everyone else getting them locally as well?

And I wonder are these tests failing because of the basic issue about clearing the lat/lng from doit volops?

@Maroo-b - I really appreciate the work you did on #335 - maybe we should pull that in here as you say?  Or should we just ensure that the volops pulled from doit do include an address?  Might that be a simpler fix?

I have to say I don't feel I've got as good a grasp of the problem as I'd like ...